src/openrct2-ui/windows/Options.cpp

```diff
@@ -636,6 +639,11 @@ static void window_options_display_mouseup(rct_window* w, rct_widgetindex widget
             config_save_default();
             w->Invalidate();
             break;
+        case WIDX_DISABLE_SCREENSAVER_LOCK:
+            gConfigGeneral.disable_screensaver_lock ^= 1;
+            config_save_default();
+            w->Invalidate();
+            break;
```

Comment:
I don't believe this is enough to actually toggle the screensaver if the relevant routine is only invoked in `CreateWindow`. Perhaps split the invocation off to a separate function, and call that function from here, too?

Reply:
After toggling the game has to be restarted, otherwise it won't have any effect.
Toggling this option here requires including SDL.h in Options.cpp file. My solution will follow soon.

Reply:
> After toggling the game has to be restarted, otherwise it won't have any effect.

Yes, it doesn't have any effect without restarting in its current state… However, if that's intended, you should at least present a warning to the user.

> Toggling this option here requires including SDL.h in Options.cpp file. My solution will follow soon.

Instead of invoking any of the SDL functions directly, you can expose a new method on e.g. the context object. (Though do keep in mind it can't be available outside of the `openrct2-ui` scope.)

Reply:
I hope the solution presented resembles what you meant.